### ISSUE TITLE
INQ: Mapping the material of walls to user defined psets

### ISSUE THREAD

**Issue by oliwerweber** [Autodesk/revit-ifc#722]
### Inquiry description

... Hi! I'm new to writing user defined psets and need help with mapping the Strucutral material under the "Material and Finishes" heading.

I guess my issue is that i dont know what parameter needs to be mapped, and what ifc element(?) to be used. Do i make sense?? 

### Revit Version

2023.1.x

### IFC for Revit Addon Version

24.x.x

### Windows Version

11 22H2

**Comment by AngelVelezSosa** [comment]
Are you trying to create a property set associated with an IFC wall for "Structural Material"?

**Comment by oliwerweber** [comment]
Yes, i need to get the "Structural Material" for walls to show up in a custom property set. 
Below is a snippet of my property set that im playing around with, along with questions. 

PropertySet:	Test	T(not sure if its type or instance?)	IfcWall
	MaterialTest					Text(im nut sure if its supposed to be Text or Label?)		Structural Material(This is where i get stuck)

**Comment by AngelVelezSosa** [comment]
PropertySet:	Test	I	IfcWall
	MaterialTest					Text      Structural Material

So it is an instance parameter because you see it on the instance, not type properties in Revit.  You can call it MaterialTest, but if you do then you have to say what the material is called in Revit, with a tab after Text.  The difference between Text and Label is that text can be 64k characters and labels only 255.

**Comment by ArtemHnatko** [comment]
### INQ: Mapping the material of walls to user defined psets - Solution Provided

Hello @oliwerweber,

Thank you for your inquiry about mapping the "Structural Material" of walls to a custom IFC property set. This is a common question when working with user-defined property sets.

Our investigation confirms that you can successfully map the "Structural Material" parameter to your custom property set. The key is to correctly identify whether the parameter is an Instance or Type parameter in Revit, and to specify the correct IFC entity.

**Solution:**

The "Structural Material" parameter for walls is a **Type parameter** in Revit. Therefore, your `DefaultUserDefinedParameterSets.txt` entry should be configured as follows:

PropertySet:	Test	T	IfcWallType
	MaterialTest	Text	Structural Material


**Explanation:**

* **`PropertySet: Test`**: Defines your custom property set named "Test".
* **`T`**: Specifies that this property set applies to **Type** entities in IFC (since "Structural Material" is a Type parameter in Revit).
* **`IfcWallType`**: Indicates that this property set should be associated with `IfcWallType` entities in the exported IFC.
* **`MaterialTest`**: This will be the name of the property in your custom IFC property set.
* **`Text`**: The data type for the exported property. "Structural Material" is a text-based value (the name of the material), so `Text` is appropriate.
* **`Structural Material`**: This is the exact name of the Revit parameter you want to map.

By using `IfcWallType` as the target IFC entity, the exporter correctly finds and maps the "Structural Material" parameter from the Revit wall type to your custom property set.

We hope this helps you achieve your desired export!

<img width="951" height="1896" alt="Image" src="https://github.com/user-attachments/assets/ec03500c-5e83-4ff9-9ffb-8cc1f08a34e2" />
